src/utils/daos_hdlr.c

```diff
@@ -1566,11 +1598,1197 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+	   	   struct file_dfs *src_file_dfs,
+	   	   struct file_dfs *dst_file_dfs)
+{
+	int rc = 0;
+
+	if (src_file_dfs->type == DAOS) {
+		rc = dfs_umount(src_file_dfs->dfs);
+		if (rc != 0) {
+			fprintf(stderr, "failed to unmount source (%d)\n", rc);
+			D_GOTO(out, rc);
+		}
+		rc = daos_cont_close(fa->src_coh, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to close source container (%d)\n", rc);
+			D_GOTO(out, rc);
+		}
+		rc = daos_pool_disconnect(fa->src_poh, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to disconnect from source pool "DF_UUIDF
+				": %s (%d)\n", DP_UUID(fa->src_p_uuid), d_errdesc(rc), rc);
+			D_GOTO(out, rc);
+		}
+	}
+	if (dst_file_dfs->type == DAOS) {
+		rc = dfs_umount(dst_file_dfs->dfs);
+		if (rc != 0) {
+			fprintf(stderr, "failed to unmount destination (%d)\n", rc);
+			D_GOTO(out, rc);
+		}
+		rc = daos_cont_close(fa->dst_coh, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to close destination container (%d)\n", rc);
+			D_GOTO(out, rc);
+		}
+		rc = daos_pool_disconnect(fa->dst_poh, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to disconnect from destination pool "DF_UUIDF
+				": %s (%d)\n", DP_UUID(fa->dst_p_uuid), d_errdesc(rc), rc);
+			D_GOTO(out, rc);
+		}
+	}
+out:
+	return rc;
+}
+
+static int fs_copy_parse_uuids(char* str,
+			       char **str_ptr,
+			       uuid_t *p_uuid,
+			       uuid_t *c_uuid,
+			       char **pool,
+			       char **cont)
+{
+	int rc = 0;
+
+	*pool = strtok_r(str, "/", str_ptr);
+	if (uuid_parse(*pool, *p_uuid) == 0) {
+		*cont = strtok_r(NULL, "/", str_ptr);
+		if ((rc = uuid_parse(*cont, *c_uuid)) != 0) {
+			fprintf(stderr, "failed to parse cont uuid\n");
+			D_GOTO(out, rc);
+		}
+	} else {
+		fprintf(stderr, "failed to parse uuid or path\n");
+		D_GOTO(out, rc = 1);
+	}	
+out:
+	return rc;
+}
+
+static int
+fs_copy_parse(char *src,
+	      char *dst,
+	      struct fs_copy_args *fa,
+	      struct file_dfs *src_file_dfs,
+	      struct file_dfs *dst_file_dfs,
+	      struct duns_attr_t *src_dattr,
+	      struct duns_attr_t *dst_dattr,
+	      char **src_str,
+	      char **src_dfs_path,
+	      int *src_str_len,
+	      char **dst_str,
+	      char **dst_dfs_path,
+	      int *dst_str_len)
+{
+	/* parse src and dst strings 
+	 * for DAOS, daos: specified in string 
+	 * for POSIX, just use path
+	 * example: --src={daos://pool/cont}
+	 * example: --dst={/a/path} */
+	int	rc		= 0;
+	char	*src_saveptr	= NULL;
+	char	*dst_saveptr	= NULL;
+	char	*src_type	= NULL;
+	char	*dst_type	= NULL;
+	char	*p_src		= NULL;
+	char	*p_dst		= NULL;
+	char	*src_pool	= NULL;
+	char	*src_cont	= NULL;
+	char	*dst_pool	= NULL;
+	char	*dst_cont	= NULL;
+
+	/* parse type, posix or daos */
+	src_type = strtok_r(src, ":", &src_saveptr);
+	p_src = strtok_r(NULL, ":", &src_saveptr);
+	dst_type = strtok_r(dst, ":", &dst_saveptr);
+	p_dst = strtok_r(NULL, ":", &dst_saveptr);
+
+	/* check for src DAOS pool/cont or UNS path */ 
+	if (strcmp(src_type, "daos") == 0) {
```

Comment:
At one point @mchaarawi recommended something like:
```
daos://<pool>/<cont>/<path>
```
This parsing doesn't seem too support the preceding `//`. Instead, it expects:
```
daos:<pool>/<cont>/<path>
```
For example:
```
> daos fs copy --src daos://$pool1/$p1cont1/source --dst dest
dfs_lookup /tmp/dbohninx/03 failed, 2
```
I'm okay with either way, but I'm not sure what is _**actually**_ expected.
If we want to enforce `daos://` exactly, maybe we can do something like this:
```
// not tested
if (strncmp(src, "daos://", 8) == 0) {
    src_type = "daos"; // Treat as daos
    src += 8; // Just move pointer to where the pool uuid should be
}
```
Then, anything else would just be treated as POSIX, including incorrect variants like these: `daos:/<pool>`, `daos:<pool>`, `daos<pool>`
In which case the POSX path probably doesn't exist, which would be caught by the other error handling.

Reply:
Also, some other error checking for this:
```
> daos fs copy --src daos:///c3501a11-697a-4bb8-a8b6-ef45cccd911a/ --dst dst
...
segmentation fault
```

Reply:
@daltonbohning I can add this if this is what we want to enforce.

Reply:
I'll let @mchaarawi comment here.

Reply:
@daltonbohning the more I think about this I think enforcing the daos:// is better because it prevents these issues/errors you are seeing better

Reply:
@daltonbohning updating this on my latest commit to enforce daos:// prefix when pool/cont are specified